018_go_list_mod_graph_why/README.md

```diff
@@ -0,0 +1,459 @@
+<!-- __JSON: gobin -m -run myitcv.io/cmd/egrunner script.sh # LONG ONLINE
+
+### Introduction
+
+`go list`, `go mod why` and `go mod graph` can be used to analyse modules and packages. In this guide we introduce each
+command as we analyse the `{{PrintOut "main module"}}` module:
+
+{{PrintOut "base repo"}}
+
+For any of the commands, see `go help` for more details, for example, `go help list` or `go help mod graph`. These help
+guides are also available [on the Go website](http://golang.org/cmd/go).
+
+### Walk-through
+
+Clone our example repo ready for analysis:
+
+```
+{{PrintBlock "setup" -}}
+```
+
+Our main module is:
+
+```
+{{PrintBlock "main module" -}}
+```
+
+The `-m` flag to `go list` causes list to list modules instead of packages.
+
+The packages in the main module are:
+
+```
+{{PrintBlock "main module pkgs" -}}
+```
+
+That is to say, the module `{{PrintOut "main module"}}` consists of the single package `{{PrintOut "pkgpath"}}`.
+
+The main module has the following structure:
+
+```
+{{PrintBlock "tree main" -}}
+```
+
+Examine the single `.go` file in the single package in our main module:
+
+```go
+{{PrintBlockOut "listmodwhygraph.go" -}}
+```
+
+Ensure the main module's `go.mod` (and `go.sum`) reflects its source code:
+
+```
+{{PrintBlock "tidy" | lineEllipsis 3 -}}
+```
+
+Building on ["Visually analysing module dependencies"](../014_mod_graph/README.md), use `go mod graph` to look at the
+module requirement graph. Remember, each of the nodes in this graph is a module (click the image for a larger version),
+and an edge `m1 -> m2` indicates that `m1` requires `m2`:
+
+![Module Dependency Graph](https://raw.githubusercontent.com/{{PrintOut "org"}}/listmodgraphwhy_analysis/master/graph.svg?sanitize=true)
```

Comment:
this graph contains text that's barely readable on my laptop. Here is a screen grab to show how tiny the text is compared to regular markdown text.
![2018-11-13-14 38 37](https://user-images.githubusercontent.com/3576549/48420224-db280380-e751-11e8-99bb-bf0e5bdc6af9.png)

Looks to me like we could make the font size larger, while keeping the image dimensions similar, unless I'm missing something.

Reply:
I was hoping to be able to ignore this problem.... But you're right, I'll try and tweak it now.

Reply:
Not a dealbreaker, so I'd drop it if fixing it is a large task :)